File: web-vue/src/types/plan.ts
import type { Workout } from './workout'

export type PlanWorkout = Workout & {
  date: string // ISO 8601 date string
}

export type PlanClient = {
  id?: string
  name: string
  imageUrl?: string
}

export type PlanStatus = 'Draft' | 'Published' | 'Revoked'

export type Plan = {
  id: string
  name: string
  clientId?: string
  client?: PlanClient
  clientName: string // Reference to client by name
  workouts: PlanWorkout[]
  status?: PlanStatus
}

File: netbackend/src/EduManage.Domain/Entities/Plan.cs
namespace EduManage.Domain.Entities;

public class Plan
{
    public required string Id { get; set; }
    public required string Name { get; set; }
    public string? Notes { get; set; }
    public required string Status { get; set; }
    public string? ClientId { get; set; }
    public required string UserId { get; set; }
    public Client? Client { get; set; }
    public ICollection<PlanWorkout> Workouts { get; set; } = [];
}
